platform/pulpcore/download/writer.py

```diff
@@ -92,6 +93,9 @@ def __init__(self, path):
         """
         super(FileWriter, self).__init__()
         self.path = path
+        self.hashers = {}
+        for hasher in hashlib.algorithms_guaranteed:
+            self.hashers[hasher] = getattr(hashlib, hasher)()
```

Comment:
For better separation of concerns, a `Writer` should not be generating digests.  This is also redundant with digests being calculated by `Validation`.  There are cleaner ways to hook this in.  For example: adding a buffer event listener to calculate all the digests to be used when creating an `Artifact`.